enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,433 @@
+      have not, and the newness could be seen as risky.
+    - However, the tangible security advantages of allowing user namespaces outweigh the theoretical
+      security risks.
+    - Plus, for the majority of users, user namespaces will not be enabled to begin with.
+
+### Drawbacks
+
+## Open Questions [optional]
+
+## Test Plan
+
+- e2e tests for a user namespaced pod, especially with different volume types to verify kernel idmapped mount support
+- long-term: e2e tests for running podman in a pod, so we have an established test path for users to know what works and what doesn't.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- All pieces of this enhancement implemented, within the TPNU feature set
+- Extensive documentation on enablement and common pitfalls
+- Unit and e2e coverage
+- Gather feedback from users rather than just developers (there are customers who are interested in trying this out)
+
+### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [Openshift-docs](https://github.com/Openshift/Openshift-docs/)
+
+### Removing a deprecated feature
+N/A
+
+## Upgrade / Downgrade Strategy
+
+- To begin with, this feature will be gated by TechPreviewNoUpgrade, so it will not be able to upgrade
+- A downgrade of the apiserver will cause this feature to not work as well, as the SCC changes will be lost on older versions
+- A downgrade of the kubelet down to 1.28 should continue to work
+    - Support for idmapped mounting was added in 1.28, so pods with volumes that have the `hostUsers` field specified will fail.
+    - Since this feature is being added in 4.17, the version skew supported goes down to 4.14, meaning it must stay in TechPreview for 4.17
+
+In the future, this feature will have the TechPreviewNoUpgrade flag removed, at which point all supported Kubelets and apiservers will
+be aware of the feature gate and attempt to create a pod with a user namespace. The only special upgrade/downgrade considerations are
+for the SCC changes, which users will loose access to if the cluster downgrades.
+
+## Version Skew Strategy
+
+In Kubernetes and Openshift, a version skew of n-3 between the kubelet and apiserver is supported. The key consideration in version skew:
+if the kube-apiserver believes the cluster supports user namespaces, will every supported kubelet create a pod with a user namespace?
+
+There is risk if this does not happen, as we intend on having the apiserver relax validation for a pod that it believes is confined by a user namespace,
+when in reality it is not, thus leading to security vulnerability.
+
+In this enhancement, we assume homogeneous feature gates, where both the kubelet and kube-apiserver have the same feature gates set.
```

Comment:
invalid assumption since old kubelets will have different featuregates and there is featuregate skew for an unlimited duration whenever the supported list changes and nodes have to roll.

Reply:
we've readded the apiserver MinimumKubeletVersion field to fix this